examples/v1/al2-custom-userdata.yaml

```diff
@@ -28,25 +28,26 @@ spec:
           operator: Gt
           values: ["2"]
       nodeClassRef:
-        apiVersion: karpenter.k8s.aws/v1beta1
+        group: karpenter.k8s.aws
         kind: EC2NodeClass
         name: al2
 ---
-apiVersion: karpenter.k8s.aws/v1beta1
+apiVersion: karpenter.k8s.aws/v1
 kind: EC2NodeClass
 metadata:
   name: al2
   annotations:
     kubernetes.io/description: "EC2NodeClass for running Amazon Linux 2 nodes with custom user data"
 spec:
-  amiFamily: AL2 # Amazon Linux 2
   role: "KarpenterNodeRole-${CLUSTER_NAME}" # replace with your cluster name
   subnetSelectorTerms:
     - tags:
         karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
   securityGroupSelectorTerms:
     - tags:
         karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
+  amiSelectorTerms:
+    - alias: al2023@latest # Amazon Linux 2023
```

Comment:
Why `AL2` is replaced with `al2023` in [al2-custom-userdata.yaml](https://github.com/aws/karpenter-provider-aws/pull/6609/files#diff-84157e726503b7287c46f3fe63cca5c29923c8bbc949bddd7cdbc9e8fc8599c9)? Does `AL2` does support custom userData?